Tests/UnitTests/EdgePublicAPITests.swift

```diff
@@ -0,0 +1,212 @@
+
+class EdgePublicAPITests: XCTestCase {
+
+    override func setUp() {
+        continueAfterFailure = false
+        EventHub.shared.start()
+    }
+
+    override func tearDown() {
+        EventHub.reset()
+    }
+
+    // MARK: Public APIs
+    func testSendEvent_xdmData_DispatchesEdgeRequest() {
+        let expectation = XCTestExpectation(description: "edge requestContent event dispatched")
+        expectation.assertForOverFulfill = true
+        MobileCore.registerEventListener(type: EventType.edge, source: EventSource.requestContent) { event in
+            let data = flattenDictionary(dict: event.data ?? [:])
+            XCTAssertEqual(1, data.count)
+            XCTAssertEqual("xdm", data["xdm.test"] as? String)
+            expectation.fulfill()
+        }
+        Edge.sendEvent(experienceEvent: ExperienceEvent(xdm: ["test": "xdm"]))
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testSendEvent_xdmDataAndData_DispatchesEdgeRequest() {
+        let expectation = XCTestExpectation(description: "edge requestContent event dispatched")
+        expectation.assertForOverFulfill = true
+        MobileCore.registerEventListener(type: EventType.edge, source: EventSource.requestContent) { event in
+            let data = flattenDictionary(dict: event.data ?? [:])
+            XCTAssertEqual(2, data.count)
+            XCTAssertEqual("xdm", data["xdm.test"] as? String)
+            XCTAssertEqual("example", data["data.rawdata"] as? String)
+            expectation.fulfill()
+        }
+        Edge.sendEvent(experienceEvent: ExperienceEvent(xdm: ["test": "xdm"], data: ["rawdata": "example"]))
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testSendEvent_xdmDataAndDatasetId_DispatchesEdgeRequest() {
+        let expectation = XCTestExpectation(description: "edge requestContent event dispatched")
+        expectation.assertForOverFulfill = true
+        MobileCore.registerEventListener(type: EventType.edge, source: EventSource.requestContent) { event in
+            let data = flattenDictionary(dict: event.data ?? [:])
+            XCTAssertEqual(2, data.count)
+            XCTAssertEqual("xdm", data["xdm.test"] as? String)
+            XCTAssertEqual("123", data["datasetId"] as? String)
+            expectation.fulfill()
+        }
+        Edge.sendEvent(experienceEvent: ExperienceEvent(xdm: ["test": "xdm"], datasetIdentifier: "123"))
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testSendEvent_xdmDataEmptyAndData_DoesNotDispatch() {
+        MobileCore.registerEventListener(type: EventType.edge, source: EventSource.requestContent) { _ in
+            XCTFail("Unexpected - event dispatched with empty xdm data")
+        }
+
+        Edge.sendEvent(experienceEvent: ExperienceEvent(xdm: [:], data: ["rawdata": "example"]))
+        sleep(1)
+    }
+
+    func testSetLocationHint_valueHint_dispatchesEdgeUpdateProperty() {
```

Comment:
I moved these test cases from the Functional test suite to the Unit Test suite. All the above code is the same as before and the below tests are new for setLocationHint and getLocationHint.